### ISSUE TITLE
neutronless.loadbalancer.test_loadbalancer.test_create_delete_basic_lb failed due to assert ONLINE coming up as false

### ISSUE THREAD

**Issue by ssorenso** [F5Networks/f5-openstack-agent#883]
* Title: neutronless.loadbalancer.test_loadbalancer.test_create_delete_basic_lb failed due to assert ONLINE coming up as false
* Attachments: 
* Details: 

##### Suggested Bug Type:
Bug

We're expecting the given service request to result in a loadbalancer being brought online, and it isn't.  This is what tests are for.

##### Traceback
```
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f9152c4eb50>
services = [{'healthmonitors': [], 'listeners': [], 'loadbalancer': {'admin_state_up': True, 'description': '', 'gre_vteps': [], ...up': True, 'description': '', 'gre_vteps': [], 'id': '50c5d54a-5a9e-4a80-9e74-8400a461a077', ...}, 'members': [], ...}]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7f91529eb710>

    def test_create_delete_basic_lb(bigip, services, icd_config, icontrol_driver):

        service_iter = iter(services)
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        env_prefix = icd_config['environment_prefix']
        fake_rpc = icontrol_driver.plugin_rpc
        hostname = bigip.get_device_name()

        folder = '%s_%s' % (env_prefix, lb_reader.tenant_id())

        # Make sure we are starting clean.
        assert not bigip.folder_exists(folder)

        # Create the loadbalancer
        lb_pending = icontrol_driver._common_service_handler(service)
        assert not lb_pending

        # Assert that update loadbalancer status was called once
        assert fake_rpc.get_call_count('update_loadbalancer_status') == 1
        call_record = fake_rpc.get_calls('update_loadbalancer_status')[0]
>       assert call_record.get("operating_status", None) == 'ONLINE'
E       AssertionError: assert 'OFFLINE' == 'ONLINE'
E         - OFFLINE
E         ?  ^^
E         + ONLINE
E         ?  ^

../neutronless/loadbalancer/test_loadbalancer.py:63: AssertionError
```

##### Impacted test instances:
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_create_delete_basic_lb`
  * Sum: `md5sum:1edafa61ba716a69ad957bd592761350`

#### OpenStack Release
Mitaka

#### Agent Version
mitaka - e46a3c98070772ff27da829e91c8f5decb822056

#### Operating System
Nightly

**Comment by ssorenso** [comment]
#885 may be closely similar to this, and am expanding this to include test(s):
* test_pool_lb_change_ratio

##### Traceback:
```
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f6c0518fe90>
services = [{'healthmonitors': [], 'l7policies': [], 'l7policy_rules': [], 'listeners': [], ...}, {'healthmonitors': [], 'l7polic...it': -1, 'default_pool_id': '75026460-addc-4edc-ba90-50bed675b54a', 'default_tls_container_id': None, ...}], ...}, ...]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7f6c04fa3bd0>

    def test_pool_lb_change_ratio(bigip, services, icd_config, icontrol_driver):
        env_prefix = icd_config['environment_prefix']
        service_iter = iter(services)
        validator = ResourceValidator(bigip, env_prefix)

        # create lb
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        folder = '{0}_{1}'.format(env_prefix, lb_reader.tenant_id())
        icontrol_driver._common_service_handler(service)
>       assert bigip.folder_exists(folder)
E       AssertionError: assert False
E        +  where False = <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f6c0518fe90>>('TEST_980e3f914f3e40359c3c2d9470fb2e8a')
E        +    where <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7f6c0518fe90>> = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f6c0518fe90>.folder_exists

../neutronless/pool/test_pool_lb_method_change.py:53: AssertionError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_pool_lb_change_ratio`
  * Sum: `md5sum:5866396773725662dde832797c8f2ac6`

**Comment by ssorenso** [comment]
Another instance:
```
request = <FixtureRequest for <Function 'test_lbaas_service_one_opflex_member'>>
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7fdf77400f90>
services = [{'healthmonitors': [], 'l7policies': [], 'l7policy_rules': [], 'listeners': [{'admin_state_up': True, 'connection_lim...n_limit': -1, 'default_pool_id': 'b67d6530-8a82-43d4-a1dc-3418f3e62411', 'default_tls_container_id': None, ...}], ...}]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7fdf77461a50>

    def test_lbaas_service_one_opflex_member(request,
                                        bigip,
                                        services,
                                        icd_config,
                                        icontrol_driver):
        env_prefix = icd_config['environment_prefix']
        service_iter = iter(services)
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        fake_rpc = icontrol_driver.plugin_rpc
        folder = '{0}_{1}'.format(env_prefix, lb_reader.tenant_id())

        lb_pending = icontrol_driver._common_service_handler(service)
>       assert bigip.folder_exists(folder)
E       AssertionError: assert False
E        +  where False = <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7fdf77400f90>>('TEST_0e79c89d48824e25bda3366280f801a5')
E        +    where <bound method BigIpClient.folder_exists of <neutronless.testlib.bigip_client.BigIpClient object at 0x7fdf77400f90>> = <neutronless.testlib.bigip_client.BigIpClient object at 0x7fdf77400f90>.folder_exists

../neutronless/member/test_single_member_opflex.py:57: AssertionError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_lbaas_service_one_opflex_member`
  * Sum: `md5sum:6b2ee7025e8b10cba72b1a7cc869a1c6`

**Comment by ssorenso** [comment]
```
bigip = <neutronless.testlib.bigip_client.BigIpClient object at 0x7f91529eb490>
services = [{'healthmonitors': [], 'listeners': [], 'loadbalancer': {'admin_state_up': True, 'description': '', 'gre_vteps': [], ...up': True, 'description': '', 'gre_vteps': [], 'id': '50c5d54a-5a9e-4a80-9e74-8400a461a077', ...}, 'members': [], ...}]
icd_config = {'AGENT': '<oslo_config.cfg.GroupAttr object at 0x3b137d0>', 'OVS': '<oslo_config.cfg.GroupAttr object at 0x3ae9410>', 'advertise_mtu': False, 'advertised_tunnel_types': ['vxlan'], ...}
icontrol_driver = <f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver.iControlDriver object at 0x7f9152bb7950>

    def test_create_delete_lb_snatoff(bigip, services, icd_config, icontrol_driver):

        service_iter = iter(services)
        service = service_iter.next()
        lb_reader = LoadbalancerReader(service)
        env_prefix = icd_config['environment_prefix']
        fake_rpc = icontrol_driver.plugin_rpc
        hostname = bigip.get_device_name()

        # Change the configuration so that Auto SNAT is used.
        icd_config['f5_snat_mode'] = "false"
        icd_config['f5_snat_addresses_per_subnet'] = 0

        folder = '%s_%s' % (env_prefix, lb_reader.tenant_id())

        # Make sure we are starting clean.
        assert not bigip.folder_exists(folder)

        # Create the loadbalancer
        icontrol_driver._common_service_handler(service)

        # Assert that update loadbalancer status was called once
        assert fake_rpc.get_call_count('update_loadbalancer_status') == 1
        call_record = fake_rpc.get_calls('update_loadbalancer_status')[0]
>       assert call_record.get("operating_status", None) == 'ONLINE'
E       AssertionError: assert 'OFFLINE' == 'ONLINE'
E         - OFFLINE
E         ?  ^^
E         + ONLINE
E         ?  ^

../neutronless/loadbalancer/test_loadbalancer_snat.py:260: AssertionError
```
* `openstack_agent_mitaka_12.1.2-overcloud.12.consoleText`
  * Type: failures
  * Test: `test_create_delete_lb_snatoff`
  * Sum: `md5sum:6d97347c24c3ed88e4310d8d156308d7`

**Comment by mattgreene** [comment]
Regression introduced with #882 and fixed with #894.
